lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -157,28 +157,42 @@ def make_cached_func_wrapper(info: CachedFuncInfo) -> Callable[..., Any]:
     some or all of the wrapper's cached values.
     """
     cached_func = CachedFunc(info)
+    return functools.update_wrapper(cached_func, info.func)
 
-    # We'd like to simply return `cached_func`, which is already a Callable.
-    # But using `functools.update_wrapper` on the CachedFunc instance
-    # itself results in errors when our caching decorators are used to decorate
-    # member functions. (See https://github.com/streamlit/streamlit/issues/6109)
```

Comment:
Previously we wrapped function with `wrapper` so solve #6109 utilizing the fact that functions in python are descriptors. 

Now we explicitly implement the descriptor protocol for `CachedFunc` (to handle `clear(arg)` logic correctly`), and we don't need to use `wrapper` function, and could return a `functools.update_wrapper` of `cached_func` as intended originally. 

See: https://realpython.com/python-descriptors/#python-descriptors-in-methods-and-functions or https://docs.python.org/3/howto/descriptor.html#functions-and-methods